bitswap/provider.go

```diff
@@ -121,27 +126,30 @@ func NewProvider(
 	}
 
 	provider := &Provider{
-		config:         config,
-		network:        network.NewFromIpfsHost(host, routing),
-		blockManager:   blockManager,
-		retriever:      retriever,
-		requestQueue:   peertaskqueue.New(peertaskqueue.TaskMerger(&overwriteTaskMerger{}), peertaskqueue.IgnoreFreezing(true)),
-		responseQueue:  peertaskqueue.New(peertaskqueue.TaskMerger(&overwriteTaskMerger{}), peertaskqueue.IgnoreFreezing(true)),
-		retrievalQueue: peertaskqueue.New(peertaskqueue.TaskMerger(&overwriteTaskMerger{}), peertaskqueue.IgnoreFreezing(true)),
+		config:                   config,
+		network:                  network.NewFromIpfsHost(host, routing),
+		blockManager:             blockManager,
+		retriever:                retriever,
+		requestQueue:             peertaskqueue.New(peertaskqueue.TaskMerger(&overwriteTaskMerger{}), peertaskqueue.IgnoreFreezing(true)),
+		requestQueueSignalChan:   make(chan struct{}, 10),
```

Comment:
so here's my recommendation for these signals: 
1. make them buffer 1
2. when writing, call:
```
select {
   case provider.requestQueueSignalChan <- struct{}{}:
   default:
}
```

Reply:
oh, nice, so if it blocks then bail, I didn't think of that!